src/components/i-calendar.vue

```diff
@@ -0,0 +1,545 @@
+    return {
+      activeDate: undefined,
+      activeMonth: null,
+      activeYear: null,
+      selectedDate: [],
+      activeMonthPicker: undefined,
+      activeYearPicker: undefined,
+      showMonthYearTable: false,
+    };
+  },
+  computed: {
+    listDays() {
+      return Array.from(Array(7), (v, i) =>
+        dayjs()
+          .day(i + 0)
+          .format('ddd'),
+      );
+    },
+    listMonth() {
+      const months = Array.from(Array(12), (v, i) => ({
+        monthDisplay: dayjs(this.activeYearPicker).month(i).format('MMM'),
+        monthValue: dayjs(this.activeYearPicker).month(i).startOf('month'),
+        idDisabled: this.checkDateDisabled(dayjs(this.activeYearPicker).month(i).endOf('month')),
+      }));
+
+      return months;
+    },
+    activeDateString() {
+      return this.activeDate ? this.activeDate.toString() : undefined;
+    },
+    activeYearPickerString() {
+      return dayjs(this.activeYearPicker.toString()).format('YYYY');
+    },
+    previousPicker() {
+      const display = [];
+
+      const previous = dayjs(this.activeDateString).date(0);
+      const current = dayjs(this.activeDateString).date(0);
+      for (let i = 0; i <= current.day(); i += 1) {
+        display.push(previous.subtract(i, 'day'));
+      }
+      return display.sort((a, b) => a.day() - b.day());
+    },
+    currentPicker() {
+      return Array.from(Array(dayjs(this.activeDateString).daysInMonth()), (v, i) => {
+        const date = dayjs(this.activeDateString).date(++i);
+        const isDisabled = this.checkDateDisabled(dayjs(date).toDate());
+        return { date, isDisabled };
+      });
+    },
+    disabledPreviousMonth() {
+      const [firstDayCurrent] = this.currentPicker;
+
+      const firstDaySubOne = dayjs(firstDayCurrent.date).subtract(1, 'day');
+      return this.checkDateDisabled(new Date(firstDaySubOne.format()));
+    },
+    disabledNextMonth() {
+      const lastDayCurrent = this.currentPicker.slice(-1).pop();
+      const lastDayDate = lastDayCurrent ? lastDayCurrent.date : undefined;
+
+      const lastDayAddOne = dayjs(lastDayDate).add(1, 'day');
+      return this.checkDateDisabled(new Date(lastDayAddOne.format()));
+    },
+  },
+  watch: {
+    activeDate: {
+      immediate: true,
+      handler(val) {
+        if (val) {
+          this.activeMonth = dayjs(val.toString()).format('MMM');
+          this.activeYear = dayjs(val.toString()).format('YYYY');
+        }
+      },
+    },
+    value: {
+      handler(val) {
+        const valueArray = [val];
+
+        const isValueSame = this.checkSame(valueArray, this.selectedDate);
+        if (!isValueSame) {
+          this.selectedDate = valueArray.filter((date) => !!date).map((date) => dayjs(date.toString()).toDate());
+        }
+      },
+    },
+    selectedDate: {
+      deep: true,
+      handler(val) {
+        const [first] = val;
+        this.$emit('onSelectDate');
+        this.$emit('input', dayjs(first.toString()).toDate());
```

Comment:
before do emit (above line 262)
check the "val" length, 
if val.length === 0
this.$emit('onSelectDate', undefined);
this.$emit('input', undefined);

AND
pass a value when $emit('onSelectDate') - > $emit('onSelectDate', dayjs(first.toString()).toDate())

Reply:
ok